### ISSUE TITLE
Only homepage cached: Page not cached because a path param was provided in the query string.

### ISSUE THREAD

**Issue by lenvanessen** [putyourlightson/craft-blitz#411]
Hi,

We have a couple of sites where only the homepage get's cached lately, and it only happens for sites that use more recent versions of Blitz.

Our blitz.log is full with lines that say:
Page not cached because a path param was provided in the query string. [https://xxx/hoe-het-werkt?p=hoe-het-werkt]

However, that's not really the url that is called, the url that is called is /hoe-het-werkt and the .htaccess file rewrites the request to the ?p=. It seems that some apache versions (like 2.4.53.) maybe confuse blitz?

Can you maybe elaborate what this check is trying to prevent in terms of unwanted behavior? It seems it wasn't there before, and the set-up where working fine for us.

**Comment by lenvanessen** [comment]
These lines here cause the problem:
https://github.com/putyourlightson/craft-blitz/blob/develop/src/services/CacheRequestService.php#L191-L195

**Comment by bencroker** [comment]
The intention for that check is to prevent caching of URLs accessed via the path param directly. So for example, the URL `https://domain.com/home` should be cached, but the URL `https://domain.com/index.php?p=home` should not. What versions of Blitz and Craft is your site running? And can you show me your Apache rewrites?

**Comment by lenvanessen** [comment]
I understand, i'm not sure why it's doing this. I can't access the domain apache config, it's a shared hosting . 
I'ts craft 4.0.2 and and blitz 4.1.2


Below is the .htaccess
```
<IfModule mod_rewrite.c>
    RewriteEngine On

    #- Blitz cache rewrite
    RewriteCond %{DOCUMENT_ROOT}/cache/blitz/%{HTTP_HOST}/%{REQUEST_URI}/index.html -s
    RewriteCond %{REQUEST_METHOD} GET
    RewriteCond %{QUERY_STRING} !token= [NC]
    RewriteRule .* /cache/blitz/%{HTTP_HOST}/%{REQUEST_URI}/index.html [L]

    # Send would-be 404 requests to Craft
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} !^/(favicon\.ico|apple-touch-icon.*\.png)$ [NC]
    RewriteRule (.+) index.php?p=$1 [QSA,L]
</IfModule>
```

**Comment by lenvanessen** [comment]
Perhaps we can access an additional check, to see if the url is actually containing the index.php string?

**Comment by bencroker** [comment]
> Perhaps we can access an additional check, to see if the url is actually containing the index.php string?

Either way, you want `https://domain.com/home` to be cached, but not `https://domain.com/?p=home`.

Can you send the live URL to info@putyourlightson.com so I can take a look at the site?

**Comment by bencroker** [comment]
I had a look at the site, but am still confused as to why your Apache redirect would return the URI `domain.com/?p=home` instead of `domain.com/index.php?p=home`. I think there may be something else at play here, but I'm not sure what it might be.

**Comment by lenvanessen** [comment]
@bencroker apache does not return the URI as domain.com/?p=home, but as domain.com/home?=home. Nevertheless, is there a workaround that can be achieved so we can get it to work? 

Perhaps check if the string contains index.php, or if the request uri matches a file in de web directory (if someone would have renamed index.php for some reason)?

**Comment by bencroker** [comment]
Well, I'd prefer to get to the root of the problem, which appears to be with the server configuration rather than the plugin, since this has not been an issue before. Do you have other sites running on Apache in which this is _not_ an issue?

**Comment by lenvanessen** [comment]
@bencroker not that I know of. But it's a DirectAdmin server, with standard configuration. I don't have any other 
I see the method is already checking for index.php, doesn't that make the path check a bit redundant since it would normally always be combined with index.php?

```
       // Ignore URIs that contain `index.php`
        if (str_contains($uri, 'index.php')) {
            Blitz::$plugin->debug('Page not cached because the URL contains `index.php`.', [], $url);

            return false;
        }
```

How would we dig a bit deeper? What would you need, i can ask the hosting guys for the apache config?

**Comment by bencroker** [comment]
Possibly, yes. I've changed the behaviour in https://github.com/putyourlightson/craft-blitz/commit/24ad8c226d9f228dffb7c81152a71dffce9236b6, can you give this a try?

**Comment by lenvanessen** [comment]
Works!

**Comment by bencroker** [comment]
Great, I'll add in the next release.

**Comment by bencroker** [comment]
Released in version 4.1.3.

**Comment by ursbraem** [comment]
🎉 🙏
